Core/Src/main.c

```diff
@@ -95,10 +103,22 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //setting the PB4 port (I think that's the GPIO port?) low
+
+	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 8, 500);
```

Comment:
When you are using `HAL_SPITransmitReceive` the fourth parameter asks for the number of bytes to be transferred. If your buffers are only holding 3 one byte (8 bits) values, why are you saying to transmit 8 bytes?